extensions/amp-selector/1.0/amp-selector.js

```diff
@@ -249,12 +250,42 @@ function OptionShim({
 function SelectorShim({
   shimDomElement,
   children,
+  form,
   multiple,
+  name,
   disabled,
   onKeyDown,
   role = 'listbox',
   tabIndex,
+  value,
 }) {
+  const input = useRef(
+    createElementWithAttributes(shimDomElement.ownerDocument, 'input', {
+      'hidden': '',
+    })
+  );
+
+  useLayoutEffect(() => {
+    const el = input.current;
+    shimDomElement.appendChild(el);
```

Comment:
Not super critical, but IMHO, it's better when a hidden input is set as a first child. At least at first. I'd not worry too much about mutations.

Reply:
Modified to set as first; could you clarify why it is better?

Reply:
I've always found it easiest to understand/debug a custom input when the hidden input is the first child. More of a stylistic reason.